packages/components/src/core/NavigationJumpTo/useIntersection.ts

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+
+/**
+ * Custom hook that observes elements to determine if they are in the viewport.
+ *
+ * @param items - Array of items to observe.
+ * @returns Object containing the visibility status of each observed item.
+ */
+export default function useInView(
+  items: Array<{
+    title: string;
+    elementRef: React.MutableRefObject<HTMLElement | null>;
+  }>
+) {
+  // If IntersectionObserver is not available (e.g., in a test environment),
+  // return a default result indicating that all elements are not in view
+  if (!window.IntersectionObserver) {
+    return items.map((item) => {
+      return {
+        el: item.elementRef,
+        isInView: false,
+      };
+    });
+  }
+
+  const [elements, setElements] = useState<{
+    [key: string]: { isInView: boolean };
+  }>({});
+
+  useEffect(() => {
+    /**
+     * Callback function called by the IntersectionObserver.
+     * Updates the visibility status of observed elements.
+     *
+     * @param entries - IntersectionObserverEntry array containing information about each observed element.
+     */
+    const observerCallback = (entries: IntersectionObserverEntry[]) => {
+      entries.forEach((entry, index) => {
+        const name = entry.target.getAttribute("id") || index;
+
+        if (entry.isIntersecting) {
+          // Element is in view
+          setElements((prev) => {
+            return {
```

Comment:
should we add a guard clause to not set element if the item is already in view?

E.g.,

```ts
setElements((prev) => {
  if (prev[name]?.isInView) return prev;

  return {
    ...prev,
    [name]: {
      el: entry,
      isInView: true,
    },
  }
})

```

Reply:
You raise a valid point. However, the IntersectionObserver callback is triggered only when there is a change in the elements intersecting with the view. As a result, there is already an inherent guard mechanism in place, similar to what you suggested. Therefore, I believe it is not necessary to explicitly include an additional guard clause in this case. :D